doc/decisions/change_tracking.md

```diff
@@ -0,0 +1,160 @@
+# Change Tracking
+
+## Problem
+
+Currently, a range of Elektra plugins are implementing some sort of change-tracking for configuration data.
+This includes, but is not limited to, the [internalnotification](../../src/plugins/internalnotification/README.md)
+and [dbus](../../src/plugins/dbus/README.md) plugins. In the near future, Elektra shall also be extended with session recording.
+
+Having multiple implementations for the same problem is problematic in multiple ways:
+
+1. pointless waste of resources, as data is duplicated in each plugin,
+2. multiplication of code, generating a maintenance burden.
+
+The current approach to change tracking in plugins is very fragile.
+Basically, what plugins currently do is:
+
+1. on `kdbGet` duplicate the returned keyset,
+2. on `kdbSet` calculate what changed in between.
+
+This basic approach will fail when there is another `kdbGet` operation for a different part of the KDB in between.
```

Comment:
I stopped reading here, as this duplicates with #4574

Reply:
I removed it and referenced #4574 instead.